ansi.go

```diff
@@ -180,8 +180,8 @@ func (p *dispatcher) CsiDispatch(s ansi.CsiSequence) {
 				p.beginBackground(fill)
 				i += 3
 			}
-		case 100, 101, 102, 103, 104, 105, 106, 107:
-			p.beginBackground(ansiPalette[v])
+		case 40, 41, 42, 43, 44, 45, 46, 47, 100, 101, 102, 103, 104, 105, 106, 107:
+			p.beginBackground(ansiPalette[v-10])
```

Comment:
Hey, why do we need the `-10` here?

Reply:
I believe to account for the color indices in https://github.com/charmbracelet/freeze/blob/main/ansi.go#L190. Since 40s and 100s are the same ANSI16 colors but for background, we need to `-10` to get the correct colors from `ansiPalette`

Reply:
Yep it's because the background and foreground colors are the same, but the background codes are shifted up by 10.

Reply:
Why don't we pull out this magic number, i.e `const backgroundColorOffset = -10`

Reply:
We could also normalize the `ansiPalette` to use indices from 0-15 which imo makes more sense

Reply:
Totally, that's _way_ clearer

Reply:
I like that idea! Or maybe something like this?
Add a separate `ansiPaletteBright`:
```go
var ansiPalette = [16]string{
	"#282a2e", // black
	"#D74E6F", // red
	"#31BB71", // green
	"#D3E561", // yellow
	"#8056FF", // blue
	"#ED61D7", // magenta
	"#04D7D7", // cyan
	"#C5C8C6", // white
}

var ansiPaletteBright = [8]string{
	"#4B4B4B", // bright black
	"#FE5F86", // bright red
	"#00D787", // bright green
	"#EBFF71", // bright yellow
	"#8F69FF", // bright blue
	"#FF7AEA", // bright magenta
	"#00FEFE", // bright cyan
	"#FFFFFF", // bright white
}
```
For foreground colors:
```go
case 30, 31, 32, 33, 34, 35, 36, 37:
	span.CreateAttr("fill", ansiPalette[v%10])
	p.lines[p.row].AddChild(span)
case 90, 91, 92, 93, 94, 95, 96, 97:
	span.CreateAttr("fill", ansiPaletteBright[v%10])
	p.lines[p.row].AddChild(span)
```
For background colors:
```go
case 40, 41, 42, 43, 44, 45, 46, 47:
	p.beginBackground(ansiPalette[v%10])
case 100, 101, 102, 103, 104, 105, 106, 107:
	p.beginBackground(ansiPaletteBright[v%10])
```